src/libs/ReportUtils.ts

```diff
@@ -4677,6 +4678,22 @@ function canBeAutoReimbursed(report: OnyxEntry<Report>, policy: OnyxEntry<Policy
     return isAutoReimbursable;
 }
 
+/**
+ * Checks if report chat contains add bank account action
+ */
+function hasAddBankAccountAction(iouReportID: string): boolean {
```

Comment:
Can we somehow deduplicate the logic with [the other place](https://github.com/Expensify/App/blob/a2f5bd5898eafc53801807c37929afaae8a23666/src/pages/home/report/ReportActionItem.js#L409)?

Reply:
Let's start with cleaning this function up.

`isSubmitterOfUnsettledReport` and `hasCreditBankAccount` are used inside the lambda provided to `find`, but they don't depend on `action.` After we clean this up, it will be clearer what is and isn't common between both use sites.

Reply:
I don't have super strong opinions (at least at this point) about how this should be refactored. But I'm sure that we should refactor **something**; there's too much copy-paste going one here.

Reply:
> I don't think so, we just duplicate 1-2 lines. I just think about create the separate function but it seems more messy

It probably depends on our creativity.

What if we pivoted, and introduced a function like this to `src/libs/ReportActionsUtils.ts` instead?

```ts
type MissingPaymentMethod = 'bankAccount' | 'wallet';

/**
 * What missing payment method does this report action indicate, if any?
 */
function getIndicatedMissingPaymentMethod(
    userWallet: UserWallet,
    reportId: string,
    reportAction: ReportAction,
): MissingPaymentMethod | undefined {
    const isSubmitterOfUnsettledReport = isCurrentUserSubmitter(reportId) && !isSettled(reportId);

    if (!isSubmitterOfUnsettledReport || reportAction.actionName !== CONST.REPORT.ACTIONS.TYPE.REIMBURSEMENTQUEUED) {
        return undefined;
    }

    const paymentType = (reportAction.originalMessage as IOUMessage)?.paymentType;

    if (paymentType === CONST.IOU.PAYMENT_TYPE.EXPENSIFY) {
        return _.isEmpty(userWallet) || userWallet.tierName === CONST.WALLET.TIER_NAME.SILVER ? 'wallet' : undefined;
    } else {
        return !store.hasCreditBankAccount() ? 'bankAccount' : undefined;
    }
}
```

We could use it in `ReportActionItem.js` in a relatively straight-forward way. We could use it in `ReportUtils.ts` like this:

```ts
function hasMissingPaymentMethod(
    userWallet: UserWallet,
    iouReportID: string,
): boolean {
    const reportActions = ReportActionsUtils.getAllReportActions(iouReportID);

    return Object.values(reportActions).some(
        (action) => ReportActionsUtils.getIndicatedMissingPaymentMethod(userWallet, iouReportID, action) !== undefined,
    );
}
```

What do you think?

Please note that this is just an inspiration; this code needs testing and, possibly, some adjustments.

Reply:
@tienifr Bump!

Reply:
That's true. This is a de-duplication/performance tradeoff. I was aware of this. I checked the implementation of the mentioned operations, and they don't seem to be complex.

On the other hand, we're talking about two UI elements that are meant to present (nearly) the exact same information. Using different functions for implementing them is a recipe for a bug where only one place is updated when the business logic changes.

Reply:
Let me know if you have some other ideas for de-duplicating this logic in another way